server/src/main/java/org/apache/druid/metadata/SQLMetadataConnector.java

```diff
@@ -824,4 +911,59 @@ public Void withHandle(Handle handle)
       log.warn(e, "Exception while deleting records from table");
     }
   }
+
+  /**
+   * Interrogate table metadata and return true or false depending on the existance of the indicated column
+   *
+   * public visibility because DerbyConnector needs to override thanks to uppercase table and column names invalidating
+   * this implementation.
+   *
+   * @param tableName The table being interrogated
+   * @param columnName The column being looked for
+   * @return boolean indicating the existence of the column in question
+   */
+  public boolean tableHasColumn(String tableName, String columnName)
+  {
+    return getDBI().withHandle(
+        new HandleCallback<Boolean>()
+        {
+          @Override
+          public Boolean withHandle(Handle handle)
+          {
+            try {
+              if (tableExists(handle, tableName)) {
+                DatabaseMetaData dbMetaData = handle.getConnection().getMetaData();
+                ResultSet columns = dbMetaData.getColumns(
+                    null,
+                    null,
+                    tableName,
+                    columnName
+                );
+                return columns.next();
+              } else {
+                return false;
+              }
+            }
+            catch (SQLException e) {
+              return false;
+            }
+          }
+        }
+    );
+  }
+
+  /**
+   * Ensure that the segment table has the proper schema required to run Druid properly.
+   *
+   * Throws RuntimeException if the column does not exist. There is no recovering from an invalid schema,
+   * the program should crash.
+   */
+  private void validateSegmentTable()
+  {
+    if (tableHasColumn(tablesConfigSupplier.get().getSegmentsTable(), "used_flag_last_updated")) {
+      return;
+    } else {
+      throw new RuntimeException("Invalid Segment Table Schema! No used_flag_last_updated column!");
```

Comment:
Would be good to link to the docs here about how to do the migration manually. We can write the link out even though it won't be valid until after the release is made: https://druid.apache.org/docs/latest/operations/upgrade-prep.html